pallets/exchange/src/lib.rs

```diff
@@ -84,41 +84,32 @@ decl_event!(
 		/// who, intention type, intention id, amount, amount sold/bought
 		IntentionResolvedAMMTrade(AccountId, IntentionType, IntentionID, Balance, Balance),
 
+		/// Intention resolved as Direct Trade
+		/// who, who - account between which direct trade happens
+		/// intention id, intention id - intentions which are being resolved ( fully or partially )
+		/// Balance, Balance  - corresponding amounts
 		IntentionResolvedDirectTrade(AccountId, AccountId, IntentionID, IntentionID, Balance, Balance),
+
+		/// Paid fees event
+		/// who, account paid to, asset, amount
 		IntentionResolvedDirectTradeFees(AccountId, AccountId, AssetId, Balance),
 
+		/// Error event - insuficient balance of specified asset
+		/// who, asset, intention type, intention id, error detail
 		InsufficientAssetBalanceEvent(AccountId, AssetId, IntentionType, IntentionID, dispatch::DispatchError),
 
-		//Note: This event can be used instead of AMMSellErrorEvent, AMMBuyErrorEvent
+		/// Intetion Error Event
+		/// who, assets, sell or buy, intention id, error detail
 		IntentionResolveErrorEvent(
```

Comment:
We now miss info about the stage at which the intention failed.

Reply:
Well, not really.  Each error event has 'error detail' which is the actual error which happened. So based on this, it is possible to tell what exactly happened. We just need to figure out how to map this error on the FE side to a meaningful/user friendly info.